specs/_features/eip7594/polynomial-commitments-sampling.md

```diff
@@ -423,8 +469,118 @@ def verify_kzg_proof_multi_impl(commitment: KZGCommitment,
     ]))
 ```
 
+#### `verify_cell_kzg_proof_batch_impl`
+
+```python
+def verify_cell_kzg_proof_batch_impl(row_commitments: Sequence[KZGCommitment],
+                                     row_indices: Sequence[RowIndex],
+                                     column_indices: Sequence[ColumnIndex],
+                                     cosets_evals: Sequence[CosetEvals],
+                                     proofs: Sequence[KZGProof]) -> bool:
+    """
+    Verify a set of cells, given their corresponding proofs and their coordinates (row_index, column_index) in the blob
+    matrix. The i-th cell is in row row_indices[i] and in column column_indices[i].
+    The list of all commitments is provided in row_commitments_bytes.
+
+    This function implements the universal verification equation introduced here:
+    https://ethresear.ch/t/a-universal-verification-equation-for-data-availability-sampling/13240
+    """
+
+    # The verification equation that we will check is pairing ( LL, LR ) = pairing (RL, [1]), where
+    # LL = sum_k r^k proofs[k],
+    # LR = [s^n]
+    # RL = RLC - RLI + RLP, with
+    #   RLC = sum_i (sum_{k_i} r^{k_i}) commitments[i]
+    #   RLI = [sum_k r^k * interpolation_poly_k(s)]
+    #   RLP = sum_k r^k * h_k^n * proofs[k]
+    #
+    # Here, the variables have the following meaning:
+    # - k < len(row_indices) is an index iterating over all cells in the input
+    # - r is a random coefficient, derived from hashing all data provided by the prover
+    # - s is the secret embedded in the KZG setup
+    # - n = FIELD_ELEMENTS_PER_CELL is the size of the evaluation domain
+    # - i ranges over all rows that are touched
+    # - k_i < len(row_indices) ranges over all cells that are in row i
+    # - interpolation_poly_k is the interpolation polynomial for the kth cell
+    # - h_k is the coset shift specifying the evaluation domain of the kth cell
+
+    # Preparation
+    num_cells = len(row_indices)
+    n = FIELD_ELEMENTS_PER_CELL
+    num_rows = len(row_commitments)
+
+    # Step 1: Derive powers of r, i.e., r^0, ..., r^{num_cells-1}
+    r = int(
+        verify_cell_kzg_proof_batch_challenge(
+            row_commitments,
+            row_indices,
+            column_indices,
+            cosets_evals,
+            proofs
+        )
+    )
+    current_power = 1
+    r_powers = []
+    for _ in range(num_cells):
+        r_powers.append(current_power)
+        current_power = (current_power * r) % BLS_MODULUS
```

Comment:
We can use [`compute_powers`](https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/polynomial-commitments.md#compute_powers) from Deneb's polynomial commitment spec here.